docs/swagger/spec.yml

```diff
@@ -321,6 +588,112 @@ definitions:
         type: string
         description: URL safe version of name
         readOnly: true
+  Layer:
+    allOf:
+      - $ref: '#/definitions/BaseModel'
+      - $ref: '#/definitions/UserTrackingMixin'
+      - type: object
+        properties:
+          name:
+            type: string
+            description: The display name of the layer
+          slugLabel:
+            type: string
+            description: URL-safe version of name
+            readOnly: true
+          description:
+            type: string
+            description: Long-form description of the layer
+          tags:
+            type: array
+            items:
+              - type: string
+          images:
+            type: string
+            description: URI to layer's image information
+            format: uri
+          thumbnails:
+            type: object
+            description: The thumbnails associated with this layer
+            properties:  # Sizes inspired by Flickr
+              large:
+                $ref: '#/definitions/TaskPopulatedField'
+              small:
+                $ref: '#/definitions/TaskPopulatedField'
+              square:
+                $ref: '#/definitions/TaskPopulatedField'
+            readOnly: true
+          boundary:
+            allOf:
+              - $ref: '#/definitions/TaskPopulatedField'
+            readOnly: true
+  LayerImage:
+    allOf:
+      - $ref: '#/definitions/BaseModel'
+      - $ref: '#/definitions/UserTrackingMixin'
+      - type: object
+        properties:
+          fileName:
+              type: string
+              description: Name of the image file
+          sourceURI:
+            type: string
+            description: URI to raw source that will be used for this image
+          metadata:
+            type: object
+            description: Metadata about this image
+          thumbnails:
+            type: object
+            description: Thumbnails associated with this layer-image
+            properties:
+              large:
+                $ref: '#/definitions/TaskPopulatedField'
+              small:
+                $ref: '#/definitions/TaskPopulatedField'
+              square:
+                $ref: '#/definitions/TaskPopulatedField'
+            readOnly: true
+          boundary:
+            allOf:
+              - $ref: '#/definitions/TaskPopulatedField'
+            readOnly: true
+        required:
+          - fileName
+          - sourceURI
+          - thumbnails
+          - boundary
+  LayerPaginated:
+    allOf:
+      - $ref: '#/definitions/PaginatedResponse'
+    properties:
+      results:
+        type: array
+        items:
+          $ref: '#/definitions/Layer'
+  LayerImagePaginated:
+    allOf:
+      - $ref: '#/definitions/PaginatedResponse'
+      - type: object
+        properties:
+          results:
+            type: array
+            items:
+              $ref: '#/definitions/LayerImage'
+  TaskPopulatedField:
```

Comment:
A field whose value is determined by the result of a task (though Tasks are not first-class citizens of the API).